temporal-sdk/src/main/java/io/temporal/internal/common/WorkflowExecutionHistory.java

```diff
@@ -33,11 +33,16 @@
 import io.temporal.common.converter.DataConverterException;
 import java.util.List;
 
+// This class by mistake leaked into a public interface and is used by users, so it can't be deleted
+// right away.
+// To fix the mistake, it was republished in a public package as
+// io.temporal.common.WorkflowExecutionHistory class
+// that extends this class.
 /**
- * Provides a wrapper with convenience methods over raw protobuf {@link History} object representing
- * workflow history
+ * @deprecated use {@link io.temporal.common.WorkflowExecutionHistory} instead.
  */
-public final class WorkflowExecutionHistory {
+@Deprecated
```

Comment:
Is there any value in keeping this class around at all? It's just internal so we can delete at will and just move everything over right?

Reply:
See my note right above the line you commented on

Reply:
See my note right above the line you commented on.
Users are already using and depending on `io.temporal.internal.common.WorkflowExecutionHistory`.
Just deleting it without deprecating will bring them to a compile time error that doesn't have a clear remedy (as users will not know where to find the migration information)

Reply:
> Users are already using and depending on [...] internal

:-( But maybe we can make _this_ internal one extend the now-public one then? And remove everything from this internal one onto the public one.

Reply:
It's possible, but this is worse and moving users in a right direction.
To don't break compile time for users in the hierarchy you propos, I will need our util methods to return INTERNAL class . This way:
1. APIs will be returning deprecated classes.
2. Users using standard IDE autocompletion will continue to get vars defined with a deprecated class.

Reply:
I don't see a reason to support two completely different classes that are essentially copy-pastes of each other.

> Unfortunately it means that you haven't really made a public type with intention to get rid of non-public type. You've made a half-public half-internal type.

I disagree with that and I don't see it like that. I don't see any problem for a user with the current structure. If we want to carry the methods explicitly into the new public class and make them delegating to the super - it can be done and it's not a big issue.